internal/querynode/search.go

```diff
@@ -39,6 +39,7 @@ func searchOnSegments(replica ReplicaInterface, segType segmentType, searchReq *
 			defer wg.Done()
 			seg, err := replica.getSegmentByID(segID, segType)
 			if err != nil {
+				errs[i] = fmt.Errorf("searchOnSegments failed, err = %s", err)
```

Comment:
If segment is released safely after listed by `validateOnStreamingReplica`, this error shall be ignored since the result is still correct.

Reply:
agree